src/components/counter.js

```diff
@@ -40,21 +34,37 @@ export class Learning_Project extends Component {
   render() {
     return (
       <View style={styles.counter_col}>
-        <View style={styles.counter_row_first}>
-          <Button
-            large
-            title="Increment +"
-            color="green"
-            onPress={this.props.actions.counterIncrement}
-          />
-          <Text style={styles.text}>{this.props.count}</Text>
-          <Button
-            large
-            title="Decrement -"
-            onPress={this.props.actions.counterDecrement}
-          />
-        </View>
-        <View style={styles.counter_row}>
+        <Button
+          large
+          title="Add counter"
+          color="blue"
+          onPress={this.props.actions.newCounter}
+        />
+        {this.props.counterList.map((counter, index) => {
```

Comment:
Using an array.map to render lists renders the whole list at once. For really large datasets this will break. It is better to use a component to lazily loads the data when needed so as not to create memory pressure. I suggest using FlatList: https://facebook.github.io/react-native/docs/flatlist.html.